engine/src/main/java/org/hibernate/search/util/impl/ClassLoaderHelper.java

```diff
@@ -69,28 +75,42 @@ private ClassLoaderHelper() {
 	 * @return a new instance of classToLoad
 	 *
 	 * @throws SearchException wrapping other error types with a proper error message for all kind of problems, like
-	 * missing proper constructor, wrong type, security errors.
+	 * missing proper constructor, wrong type, securitymanager errors.
 	 */
 	public static <T> T instanceFromClass(Class<T> targetSuperType, Class<?> classToLoad, String componentDescription) {
 		checkClassType( classToLoad, componentDescription );
-		checkHasNoArgConstructor( classToLoad, componentDescription );
-		Object instance;
+		final Constructor<?> constructor = getNoArgConstructor( classToLoad, componentDescription );
+		final Object instance = untypedInstanceFromClass( classToLoad, componentDescription );
+		return verifySuperTypeCompatibility( targetSuperType, instance, classToLoad, componentDescription );
+	}
+
+	/**
+	 * Creates an instance of target class. Similar to {@link #instanceFromClass(Class, Class, String)} but not checking
+	 * the created instance will be of any specific type: using {@link #instanceFromClass(Class, Class, String)} should
+	 * be preferred whenever possible.
+	 *
+	 * @param <T> the type of targetSuperType: defines the return type
+	 * @param classToLoad the class to be instantiated
+	 * @param componentDescription a role name/description to contextualize error messages. Ideally should be provided, but it can handle null.
+	 *
+	 * @return a new instance of classToLoad
+	 *
+	 * @throws SearchException wrapping other error types with a proper error message for all kind of problems, like
+	 * missing proper constructor, securitymanager errors.
+	 */
+	public static <T> T untypedInstanceFromClass(final Class<T> classToLoad, final String componentDescription) {
```

Comment:
Ideally, shouldn't you call `checkClassType( classToLoad, componentDescription );` here?

Reply:
+1 nice improvement